.github/workflows/test.yml

```diff
@@ -73,73 +86,110 @@ jobs:
             cbbr:
               - products/cbbr/**
               - python/**
-            cpdb:
-              - products/cpdb/**
-              - python/**
             checkbook:
               - products/checkbook/**
               - python/**
-            facilities:
-              - products/facilities/**
-              - python/**
-            knownprojects:
-              - products/knownprojects/**
-              - python/**
             zap:
               - products/zap-opendata/**
               - python/**
-      - name: Confirm changes
+      - name: Confirm changes and set matrix
+        id: set-matrix
         run: |
           echo "path filters with changed files: ${{ steps.filter.outputs.changes }}"
-  dcpy:
-    needs: check_changes
-    if: github.event_name != 'pull_request' || contains(needs.check_changes.outputs.path_filters, 'dcpy')
-    uses: ./.github/workflows/dcpy_test.yml
-    secrets: inherit
-  build:
+          filter=$(echo '${{ steps.filter.outputs.changes }}' | sed 's/[][]//g' | sed -r 's/(\w+)(,|$)/"\1"\2/g')
+          matrix=$(cat .github/workflows/data/pytest.json | jq "map(. | select(.name|IN(${filter})))")
+          echo "matrix={\"include\":$(echo $matrix)}" >> $GITHUB_OUTPUT
```

Comment:
So basic logic is on 99 we clean the output of the filter step to better feed it into line 100.

Line 100 we take the json data from `pytest.json` and filter it to entries which have changes (based on `filter` variable). We then export this to github step output so we can set the matrix to this json in line 115